sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1275,29 +1275,15 @@ class Dataset[T] private[sql](
     require(colNames.size == cols.size,
```

Comment:
Is this check still needed?

Reply:
I think it's still good to have to be fail-fast, especially for connect?